application/client/src/app/ui/views/toolbar/search/input/component.ts

```diff
@@ -83,16 +89,30 @@ export class ViewSearchInput
 
     public ngAfterViewInit(): void {
         this.input.bind(this.searchInputRef.nativeElement, this.recentPanelRef);
-        this.input.actions.accept.subscribe(() => {
-            const filter = this.input.asFilter();
-            this.recent.update(filter.filter);
+        this.input.actions.accept.subscribe((flagUpdate: boolean) => {
+            let filter: IFilter;
+            if (flagUpdate) {
+                if (this.active === undefined) {
+                    return;
+                }
+                filter = {
+                    filter: this.active.filter.filter,
+                    flags: this.input.flags,
+                };
+            } else {
+                filter = this.input.asFilter();
+                this.recent.update(filter.filter);
+            }
```

Comment:
Okay, I see what you want to achieve here. But I think you've found a more fundamental mistake in the solution. `this.recent.update(filter.filter);` should not happen in component, it should happen in a place of search triggering, in `this.session.search.state().setActive(filter)`. Why? Because we can trigger search from different places - context menu, from here and others. 

Well, I would suggest leaving the event `accept` without changes, but move the saving of recent to the executor of search - it will put things into the right places.